File: generated_code/optimized/balanced/java/8.java
import java.util.*;

class Solution {
    public List<Integer> sumProduct(List<Integer> numbers) {
        int sum = 0;
        int product = 1;

        if (numbers instanceof RandomAccess) {
            for (int i = 0, n = numbers.size(); i < n; i++) {
                int v = numbers.get(i);
                sum += v;
                product *= v;
            }
        } else {
            for (Integer v : numbers) {
                sum += v;
                product *= v;
            }
        }

        return List.of(sum, product);
    }
}

File: generated_code/optimized/runtime/detailed/cpp/8.cpp
#include <vector>
using namespace std;

vector<int> sum_product(vector<int> numbers) {
    int sum = 0;
    int product = 1;

    const int* p = numbers.data();
    const int* end = p + numbers.size();

    for (; p != end; ++p) {
        sum += *p;
        product *= *p;
    }

    return {sum, product};
}
